omnigibson/objects/usd_object.py

```diff
@@ -86,6 +86,33 @@ def __init__(
             **kwargs,
         )
 
+    def prebuild(self, stage):
+        # Load the object into the given USD stage
+        usd_path = self._usd_path
+        if self._encrypted:
+            # Create a temporary file to store the decrytped asset, load it, and then delete it
+            encrypted_filename = self._usd_path.replace(".usd", ".encrypted.usd")
+            decrypted_fd, usd_path = tempfile.mkstemp(os.path.basename(self._usd_path), dir=og.tempdir)
+            os.close(decrypted_fd)
+            decrypt_file(encrypted_filename, usd_path)
+
+        # object_stage = lazy.pxr.Usd.Stage.Open(usd_path)
+        # root_prim = object_stage.GetDefaultPrim()
+
+        # The /World in the scene USD will be mapped to /World/scene_i in Isaac Sim.
+        prim_path = "/World" + self._relative_prim_path
+
+        # TODO: maybe deep copy the prim tree EXCEPT the visual meshes. How?
+        prim = stage.GetPrimAtPath(prim_path)
+        if not prim.IsValid():
+            prim = stage.DefinePrim(prim_path, "Xform")
+        assert prim.GetReferences().AddReference(usd_path)
+
+        # TODO: After we switch to a deep copy, we can enable this to remove the temporary file
+        # del object_stage
+        # if self._encrypted:
+        #     os.remove(usd_path)
```

Comment:
clean up these function? Lots of TODOs and unused code.

Reply:
To be followed up.